pkg/publish/ingress.go

```diff
@@ -176,74 +186,97 @@ func Ingress(req router.Request, svc *v1.ServiceInstance) (result []kclient.Obje
 					if err != nil {
 						return nil, err
 					}
-					targets[hostname] = Target{Port: port.TargetPort, Service: svc.Name}
-					rules = append(rules, getIngressRule(svc, hostname, port.Port))
+					clusterDomainTargets[hostname] = Target{Port: port.TargetPort, Service: svc.Name}
+					clusterDomainRules = append(clusterDomainRules, getIngressRule(svc, hostname, port.Port))
 				}
 			}
 		} else {
 			if len(ports) > 1 {
 				return nil, fmt.Errorf("multiple ports bound to the same hostname [%s]", hostname)
 			}
-			targets[hostname] = Target{Port: ports[0].TargetPort, Service: svc.Name}
-			rules = append(rules, getIngressRule(svc, hostname, ports[0].Port))
+			customDomainTargets[hostname] = Target{Port: ports[0].TargetPort, Service: svc.Name}
+			customDomainRules = append(customDomainRules, getIngressRule(svc, hostname, ports[0].Port))
 		}
 	}
 
-	if len(rules) == 0 {
-		return
-	}
+	for _, rules := range []struct {
+		rules []networkingv1.IngressRule
+		name  string
+	}{
+		{rules: clusterDomainRules, name: clusterDomain},
+		{rules: customDomainRules, name: customDomain},
+	} {
+		if len(rules.rules) == 0 {
+			continue
+		}
+		var targets map[string]Target
+		if rules.name == clusterDomain {
+			targets = clusterDomainTargets
+		} else {
+			targets = customDomainTargets
+		}
```

Comment:
Couldn't you put targets in the struct you're looping over and drop this condition and use `rules.targets`?

Reply:
yep good idea!